e2e/isolated/isolated_test.go

```diff
@@ -0,0 +1,247 @@
+			Expect("./copilot").Should(BeADirectory())
+		})
+		It("app ls includes new app", func() {
+			Eventually(cli.AppList, "30s", "5s").Should(ContainSubstring(appName))
+		})
+		It("app show includes app name", func() {
+			appShowOutput, err := cli.AppShow(appName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(appShowOutput.Name).To(Equal(appName))
+			Expect(appShowOutput.URI).To(BeEmpty())
+		})
+	})
+
+	Context("when deploying resources to be imported", func() {
+		BeforeAll(func() {
+			err := aws.CreateStack(vpcStackName, vpcStackTemplatePath)
+			Expect(err).NotTo(HaveOccurred(), "create vpc cloudformation stack")
+			err = aws.WaitStackCreateComplete(vpcStackName)
+			Expect(err).NotTo(HaveOccurred(), "vpc stack create complete")
+		})
+
+		It("parse vpc stack output", func() {
+			outputs, err := aws.VPCStackOutput(vpcStackName)
+			Expect(err).NotTo(HaveOccurred(), "get VPC stack output")
+			for _, output := range outputs {
+				switch output.OutputKey {
+				case "PrivateSubnets":
+					vpcImport.PrivateSubnetIDs = output.OutputValue
+				case "VpcId":
+					vpcImport.ID = output.OutputValue
+				}
+			}
+			if vpcImport.ID == "" || vpcImport.PrivateSubnetIDs == "" {
+				err = errors.New("resources are not configured properly")
+			}
+			Expect(err).NotTo(HaveOccurred(), "invalid vpc stack output")
+		})
+	})
+
+	Context("when adding environment with imported vpc resources", func() {
+		var testEnvInitErr error
+		BeforeAll(func() {
+			_, testEnvInitErr = cli.EnvInit(&client.EnvInitRequest{
+				AppName:       appName,
+				EnvName:       envName,
+				Profile:       "default",
+				Prod:          false,
+				VPCImport:     vpcImport,
+				CustomizedEnv: true,
+			})
+		})
+
+		It("env init should succeed for 'private' env", func() {
+			Expect(testEnvInitErr).NotTo(HaveOccurred())
+		})
+		It("env ls should list private env", func() {
+			envListOutput, err := cli.EnvList(appName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(envListOutput.Envs)).To(Equal(1))
+			env := envListOutput.Envs[0]
+			Expect(env.Name).To(Equal(envName))
+			Expect(env.Prod).To(BeFalse())
+			Expect(env.ExecutionRole).NotTo(BeEmpty())
+			Expect(env.ManagerRole).NotTo(BeEmpty())
+		})
+	})
+
+	Context("when creating and deploying a backend service in private subnets", func() {
+		var initErr error
+		BeforeAll(func() {
+			_, initErr = cli.SvcInit(&client.SvcInitRequest{
+				Name:       svcName,
+				SvcType:    "Backend Service",
+				Dockerfile: "./backend/Dockerfile",
+				SvcPort:    "80",
+			})
+		})
+
+		It("should not return an error", func() {
+			Expect(initErr).NotTo(HaveOccurred())
+		})
+		It("svc init should create a svc manifest", func() {
+			Expect("./copilot/backend/manifest.yml").Should(BeAnExistingFile())
+		})
+		It("should write 'http' and private placement to the manifest", func() {
+			f, err := os.OpenFile("./copilot/backend/manifest.yml", os.O_WRONLY|os.O_APPEND, 0644)
```

Comment:
nit: this definitely works, no need to change - I just wanted to point out an alternative. If you want you can also add an existing `copilot/backend/manifest.yml` file so that the CLI doesn't create a new one. This way you don't need to write the string at the end of the file :)

Reply:
Oh, I like that more. Changing!